lib/peopleHR.ts

```diff
@@ -71,6 +72,81 @@ async function parseXml(xmlString: string): Promise<any> {
     );
 }
 
+/**
+ * Pulls the PeopleHR RSS feed data from the cache,
+ * or fetches it from the feed if the cache has expired
+ * @returns JSON of the RSS feed
+ */
+async function getJobPostingData() {
+    const cacheFilePath = path.join(
+        process.cwd(),
+        'files',
+        'peopleHRcache.json',
+    );
+
+    if (existsSync(cacheFilePath)) {
+        const cacheFile = readFileSync(cacheFilePath, 'utf8');
+        const cacheJSON = JSON.parse(cacheFile);
+        const cacheTTL = 10 * 1000; //One hour 60 * 60 * 1000
+
+        if (cacheJSON.lastUpdated + cacheTTL < Date.now()) {
+            // The cache has timed out, fetch new data from PeopleHR
+
+            const peopleHRJobPostingsXML = await fetchPeopleHRFeed();
+
+            if (peopleHRJobPostingsXML === null) {
+                // In the event of a server error, don't update the job listings
+                // if we have cached listings still available to use.
+
+                const fileData = {
+                    lastUpdated: Date.now(),
+                    jobJSON: cacheJSON.jobJSON,
+                };
+
+                writeFileSync(cacheFilePath, JSON.stringify(fileData));
+
+                return cacheJSON.jobJSON;
+            } else {
+                // If we've received a 200 code from the server, update according to the feed contents
+                const peopleHRJobPostings = await parseXml(
```

Comment:
**Question**: Theres a snippet in the parseXml function I've copied and pasted below which rejects the promise incase of an error. 
I wonder if we want to log a reason if its rejected? I'm not as familiar with logging errors in Node/Sentry compared to Django, but I would think this could error isn't being caught any where and requires a catch method within parseXml or here
i.e: `const peopleHRJobPostings = await parseXml(peopleHRJobPostingsXML).catch((e) => {console.error("Error: ", e)});`


```
// snippet from parseXml
if (err) {
    reject(err);
}
```

Reply:
Good point, updated from rejecting the error to throwing the error and catching it after `fetchPeopleHRFeed` is used in the code as you've suggested below.